src/encoding.ts

```diff
@@ -115,6 +120,24 @@ export function decode (text: string | number = ''): string {
   }
 }
 
+/**
+ * Decode query value (taking care to handle vue-router substitutions).
+ *
+ * @param text - string to decode
+ * @returns decoded string
+ */
+export function decodeQueryValue (text: string): string {
+  return decode(
+    text
+      // Decode plus as space
+      .replace(PLUS_RE, encodeURIComponent(' '))
```

Comment:
I think simple `PLUS_RE, ' '` is enough (and tests pass without rest and encoding inside decode fn...